website/documentation/guides/monitoring_and_troubleshooting/analysing-node-failures/_index.md

```diff
@@ -0,0 +1,68 @@
+If a node was getting unhealthy, the last conditions can be found in the logs of the `machine-controller-manager` by using a query like `{pod_name=~"machine-controller-manager.*"}|="<node-name>"`.
+
+**Caveat**: every `node` resource is backed by a corresponding `machine` resource managed by machine-controller-manager. Usually two corresponding `node` and `machine` resources have the same name with the exception of AWS. Here you first need to find with the above query the corresponding `machine` name, typically via a log like this
+```
+2023-04-05 12:02:08	{"log":"Conditions of Machine \"shoot--demo--cluster-pool-z1-6dffc-jh4z4\" with providerID \"aws:///eu-central-1/i-0a6ad1ca4c2e615dc\" and backing node \"ip-10-55-138-185.eu-central-1.compute.internal\" are changing","pid":"1","severity":"INFO","source":"machine_util.go:629"}
+```
+This reveals that `node` `ip-10-55-138-185.eu-central-1.compute.internal` is backed by `machine` `shoot--demo--cluster-pool-z1-6dffc-jh4z4`. On infrastructures other than AWS you can omit this step.
+
+With the machine name at hand, now search for log entries with `{pod_name=~"machine-controller-manager.*"}|="<machine-name>"`.
+In case the node had failing conditions, you'd find logs like this:
+```
+2023-04-05 12:02:08	{"log":"Machine shoot--demo--cluster-pool-z1-6dffc-jh4z4 is unhealthy - changing MachineState to Unknown. Node conditions: [{Type:ClusterNetworkProblem Status:False LastHeartbeatTime:2023-04-05 11:58:39 +0000 UTC LastTransitionTime:2023-03-23 11:59:29 +0000 UTC Reason:NoNetworkProblems Message:no cluster network problems} ... {Type:Ready Status:Unknown LastHeartbeatTime:2023-04-05 11:55:27 +0000 UTC LastTransitionTime:2023-04-05 12:02:07 +0000 UTC Reason:NodeStatusUnknown Message:Kubelet stopped posting node status.}]","pid":"1","severity":"WARN","source":"machine_util.go:637"}
+```
+
+In the example above, the reason for an unhealthy node was that `kubelet` failed to renew its heartbeat.
+Whatever the failure condition might have been, if a node gets unhealthy, it will be terminated by `machine-controller-manager` after the `machineHealthTimeout` has elapsed (this parameter can be configured in your [shoot spec](https://github.com/gardener/gardener/blob/v1.68.0/example/90-shoot.yaml#L132))
+
+### Check the Node Logs
+For each `node` the kernel and `kubelet` logs as well as a few others are scraped and can be queried with this query `{nodename="<node-name>"}`
+This might reveal OS specific issues or in the absence of any logs (e.g. after the node went unhealthy) might indicate a network disruption.
+Infrastructure provides might be able to provide more information on particular VM failures in such cases.
+
+### Check the Network Problem Detector Dashboard
+If your Gardener installation utilizes https://github.com/gardener/gardener-extension-shoot-networking-problemdetector, you can check the dashboard named "Network Problem Detector" in Grafana for hints on network issues on the node of interest.
```

Comment:
```suggestion
If your Gardener installation utilizes [
gardener-extension-shoot-networking-problemdetector](https://github.com/gardener/gardener-extension-shoot-networking-problemdetector), you can check the dashboard named "Network Problem Detector" in Grafana for hints on network issues on the node of interest.
```

Reply:
Is that really required? For our internal doc tooling? I see that github and actually all local md tools I used so far render strings starting with `http://` or `https://` always as links without the need for square/round brackets.

Reply:
This change is aimed to improve the readability of the information - it looks better as a clickable link than as a url.

Reply:
Hmm, I thought you suggested yesterday to just put brackets without adding a custom link name, like [https://...](https://...), ok, so your argument is the custom link text, sure, why not (although I wouldn't agree that it necessarily improves readability, often I want to convey the information e.g. where a certain project is located (e.g. on public github, part of the Gardener org) without the user clicking or hovering the link) but I don't mind changing it.